akka-http-core/src/main/scala/akka/http/impl/model/parser/UriParser.scala

```diff
@@ -255,6 +255,33 @@ private[http] final class UriParser(val input: ParserInput,
       case Left(error) => fail(error, "request-target")
     }
 
+  /////////////////////////// ADDITIONAL HTTP/2-SPECIFIC RULES /////////////////////////
+
+  // https://tools.ietf.org/html/rfc7540#section-8.1.2.3
+  // https://tools.ietf.org/html/rfc3986#section-3.2 - without deprecated userinfo
+  def `http2-authority-pseudo-header` = hostAndPort
+
+  def parseHttp2AuthorityPseudoHeader(): Uri.Authority =
+    rule(`http2-authority-pseudo-header` ~ EOI).run() match {
+      case Right(_) =>
+        Authority(_host, _port)
+      case Left(error) => fail(error, "http2-authority-pseudo-header")
+    }
+
+  // https://tools.ietf.org/html/rfc7540#section-8.1.2.3
+  def `http2-path-pseudo-header` = rule(
+    `absolute-path` ~ optional('?' ~ rawQueryString) // origin-form
+        // TODO: asterisk-form
```

Comment:
Good point, do we support asterisk form anywhere yet?

Reply:
Asterisk form is mentioned here https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/model/parser/UriParser.scala#L248 but not actually parsed.

The question I had is how asterisk form would be represented in an `HttpRequest`. Asterisk form doesn't really confirm to a `Uri` or a `Uri.Path`.

Current:

```scala
final class HttpRequest(
  ...
  val uri:      Uri,
```

New?

```scala
sealed trait RequestTarget
case class UriTarget(uri: Uri) extends RequestTarget
case object AsteriskTarget extends RequestTarget

final class HttpRequest(
  ...
  val target:      RequestTarget,
  ...
) {
  ...
  def uri: Uri = target match {
    case UriTarget(uri) => uri
    case AsteriskTarget => throw new UnsupportedOperationException(...)
  }
```

Reply:
BTW, I would prefer to leave asterisk form out of this PR to keep the scope of this PR contained. It could be something to do in the future.

Reply:
Yes, definitely. I guess it might get quite hard to fit asterisk form in in a backwards compatible way.